compositor_render/src/state/render_loop.rs

```diff
@@ -91,31 +91,74 @@ pub(super) fn read_outputs(
+                OutputFrameFormat::PlanarYuv420Bytes => {
+                    let (y, u, v) = RGBColor::BLACK.to_yuv();
+                    ctx.wgpu_ctx.utils.fill_r8_with_value(
+                        ctx.wgpu_ctx,
+                        output.output_texture.yuv_textures().plane(0),
+                        y,
+                    );
+                    ctx.wgpu_ctx.utils.fill_r8_with_value(
+                        ctx.wgpu_ctx,
+                        output.output_texture.yuv_textures().plane(1),
+                        u,
+                    );
+                    ctx.wgpu_ctx.utils.fill_r8_with_value(
+                        ctx.wgpu_ctx,
+                        output.output_texture.yuv_textures().plane(2),
+                        v,
+                    );
+
+                    let pending_download = output.output_texture.start_download(ctx.wgpu_ctx);
+                    partial_textures.push(PartialOutputFrame::PendingYuvDownload {
+                        output_id: output_id.clone(),
+                        pending_download,
+                        resolution: output.output_texture.resolution().to_owned(),
+                    });
+                }
+                OutputFrameFormat::RgbaWgpuTexture => {
+                    let output_resolution = output.output_texture.resolution();
+                    let wgpu_texture =
+                        ctx.wgpu_ctx
+                            .device
+                            .create_texture(&wgpu::TextureDescriptor {
+                                label: None,
+                                size: wgpu::Extent3d {
+                                    width: output_resolution.width as u32,
+                                    height: output_resolution.height as u32,
+                                    depth_or_array_layers: 1,
+                                },
+                                mip_level_count: 1,
+                                sample_count: 1,
+                                dimension: wgpu::TextureDimension::D2,
+                                format: wgpu::TextureFormat::Rgba8Unorm,
+                                usage: wgpu::TextureUsages::COPY_SRC
+                                    | wgpu::TextureUsages::COPY_DST
+                                    | wgpu::TextureUsages::TEXTURE_BINDING
+                                    | wgpu::TextureUsages::RENDER_ATTACHMENT,
+                                view_formats: &[wgpu::TextureFormat::Rgba8Unorm],
+                            });
+                    let view = wgpu_texture.create_view(&wgpu::TextureViewDescriptor::default());
+                    let compositor_texture = crate::wgpu::texture::Texture {
+                        texture: wgpu_texture,
+                        view,
+                    };
+                    ctx.wgpu_ctx
+                        .utils
+                        .fill_r8_with_value(ctx.wgpu_ctx, &compositor_texture, 0.0);
```

Comment:
transparent is fine, no strong opinion

Reply:
The drawback is that YUV returns black in such cases. It would be weird to return different output content for different output types (with the same scene).

Only the fack that RGBA supports transparency and YUV doesn't make this somewhat justifiable.

Reply:
this is not public API, it's different output type, I think it's fine to skip unnecessary operations